File: ScriptGraph/TemplateScriptNode/GestureIndexFingerUp/CGIndexFingerUp.lua
local CGIndexFingerUp = CGIndexFingerUp or {}
CGIndexFingerUp.__index = CGIndexFingerUp

function CGIndexFingerUp.new()
    local self = setmetatable({}, CGIndexFingerUp)
    self.inputs = {}
    self.nexts = {}
    self.handAction = ""
    return self
end

function CGIndexFingerUp:setNext(index, func)
    self.nexts[index] = func
end

function CGIndexFingerUp:setInput(index, func)
    self.inputs[index] = func
end

function CGIndexFingerUp:update()
    if self.nexts[0] == nil and self.nexts[1] == nil and self.nexts[2] == nil then
        return
    end

    local result = Amaz.Algorithm.getAEAlgorithmResult()
    local hand = result:getHandInfo(self.inputs[0]())
    if hand == nil then
        return
    end

    local has_action = hand.action == Amaz.HandAction.INDEX_FINGER_UP
    if self.handAction ~= "INDEX_FINGER_UP" and has_action then
        self.handAction = "INDEX_FINGER_UP"
        if self.nexts[0] then
            self.nexts[0]()
        end
        if self.nexts[1] then
            self.nexts[1]()
        end
    elseif self.handAction == "INDEX_FINGER_UP" and has_action then
        self.handAction = "INDEX_FINGER_UP"
        if self.nexts[0] then
            self.nexts[0]()
        end
    elseif self.handAction == "INDEX_FINGER_UP" and not has_action then
        self.handAction = ""
        if self.nexts[2] then
            self.nexts[2]()
        end
    end
end
return CGIndexFingerUp

File: ScriptGraph/TemplateScriptNode/GestureIndexFingerUp/CGIndexFingerUp.js
/**
 * @file CGIndexFingerUp.js
 * @author Weifeng
 * @date 2022-01-04
 * @brief Detect index finger up gesture
 * @copyright Copyright (c) 2022, ByteDance Inc, All Rights Reserved
 */

const {CGGestureDetectionBase} = require('./CGGestureDetectionBase');
const Amaz = effect.Amaz;

class CGIndexFingerUp extends CGGestureDetectionBase {
  constructor() {
    super();
    this.action = Amaz.HandAction.INDEX_FINGER_UP;
    this.twoHandGesture = false;
  }
}

exports.CGIndexFingerUp = CGIndexFingerUp;
